### PR TITLE
:sparkles: Create the main data structure

### LINKED ISSUE #9: :bricks: Create the main data structure
# 👮‍♂️ What's needed
We need to have a commune data structure between the different files format to easily convert one format to another. This data structure will be filled by all the data of the document when we will parse it. 

# 📜 Sources
image

### PR DESCRIPTION
**Pull request by Marius-P1**
I added the main data structure for storing all the content of a Document, based on :
image

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ pipeline {
         // Set the ssh key for the mirror using secret private key
         PRIVATE_KEY = credentials('EPITECH_SSH_KEY')
         PUBLIC_KEY = credentials('EPITECH_SSH_PUBKEY')
-        MIRROR_URL = 'git@github.com:EpitechPromo2027/B-FUN-400-NAN-4-1-compressor-axel.eckenberg.git'
+        MIRROR_URL = 'https://github.com/EpitechPromo2027/B-FUN-400-NAN-4-1-mypandoc-simon.ganier-lombard'
```
Comment (axel-eck) [inline Jenkinsfile]: The mirror is made using the ssh key so the url must be in ssh format for the mirrror to succeed

**Review by landryarki — APPROVED** [review]
:+1:

File: examples/syntaxe.json [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                ]
+            }
+        }
+    ]
+}
```
Comment (axel-eck) [inline examples/syntaxe.json:114]: Is the missing trailing new line intended, and we shall not add one in the result or is this a typo ?

**Review by axel-eck — CHANGES_REQUESTED** [review]
Please update the mirror url, then it's all good

**Review by 6im0n — APPROVED** [review]
Good for me ! :+1:

File: examples/syntaxe.json (hunk 2 above)
Comment (Marius-P1) [reply examples/syntaxe.json:114]: This is the exact file in the examples given by Epitech but as Hugo say in the review, the transformation into pretty JSON is just a bonus and not a mandatory

File: Jenkinsfile [hunk 3]
```diff
@@ -4,7 +4,7 @@ pipeline {
         // Set the ssh key for the mirror using secret private key
         PRIVATE_KEY = credentials('EPITECH_SSH_KEY')
         PUBLIC_KEY = credentials('EPITECH_SSH_PUBKEY')
-        MIRROR_URL = 'git@github.com:EpitechPromo2027/B-FUN-400-NAN-4-1-compressor-axel.eckenberg.git'
+        MIRROR_URL = 'https://github.com/EpitechPromo2027/B-FUN-400-NAN-4-1-mypandoc-simon.ganier-lombard.git'
```
Comment (axel-eck) [inline Jenkinsfile]: Please use this format, the mirror will not work else.
```suggestion
        MIRROR_URL = 'git@github.com:EpitechPromo2027/B-FUN-400-NAN-4-1-mypandoc-simon.ganier-lombard.git'
```

**Review by axel-eck — CHANGES_REQUESTED** [review]
The mirror url is still not good

**Review by axel-eck — APPROVED** [review]

**Pull request merged**